src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/unsubscribe_button_request.cc

```diff
@@ -93,35 +94,67 @@ void UnsubscribeButtonRequest::Run() {
     return;
   }
 
-  if (!app->UnsubscribeFromButton(
-          static_cast<mobile_apis::ButtonName::eType>(btn_id))) {
-    SDL_LOG_ERROR("App doesn't subscribe to button " << btn_id);
+  if (!app->IsSubscribedToButton(btn_id)) {
+    SDL_LOG_ERROR("App is not subscribed to button " << btn_id);
     SendResponse(false, mobile_apis::Result::IGNORED);
     return;
   }
 
-  SendUnsubscribeButtonNotification();
-  SendResponse(true, mobile_apis::Result::SUCCESS);
+  (*message_)[str::msg_params][str::app_id] = app->app_id();
+  StartAwaitForInterface(HmiInterfaces::HMI_INTERFACE_Buttons);
+  SendHMIRequest(hmi_apis::FunctionID::Buttons_UnsubscribeButton,
+                 &(*message_)[app_mngr::strings::msg_params],
+                 true);
+}
+
+void UnsubscribeButtonRequest::on_event(const event_engine::Event& event) {
+  SDL_LOG_AUTO_TRACE();
+  using namespace helpers;
+
+  const smart_objects::SmartObject& message = event.smart_object();
+
+  if (hmi_apis::FunctionID::Buttons_UnsubscribeButton != event.id()) {
+    SDL_LOG_ERROR("Received unknown event.");
+    return;
+  }
+  EndAwaitForInterface(HmiInterfaces::HMI_INTERFACE_Buttons);
+  ApplicationSharedPtr app =
+      application_manager_.application(CommandRequestImpl::connection_key());
+
+  if (!app) {
+    SDL_LOG_ERROR("NULL pointer.");
+    return;
+  }
+
+  hmi_apis::Common_Result::eType hmi_result =
+      static_cast<hmi_apis::Common_Result::eType>(
+          message[strings::params][hmi_response::code].asInt());
+  std::string response_info;
+  GetInfo(message, response_info);
+  const bool result = PrepareResultForMobileResponse(
+      hmi_result, HmiInterfaces::HMI_INTERFACE_Buttons);
+
+  if (result) {
+    const mobile_apis::ButtonName::eType btn_id =
+        static_cast<mobile_apis::ButtonName::eType>(
+            (*message_)[str::msg_params][str::button_name].asInt());
+    app->UnsubscribeFromButton(
+        static_cast<mobile_apis::ButtonName::eType>(btn_id));
+  }
```

Comment:
It seems that this code block is redundant, because we've already unsubscribe app from button in on_event() method of HMI version of this request

Reply:
removed in 2529ff7558e0a8cc1429308d019fd9e9fd7b57ca